main/display/display.c

```diff
@@ -378,35 +354,146 @@ void transformPixel(int16_t* x, int16_t* y, int16_t transX,
 void drawWsg(display_t* disp, wsg_t* wsg, int16_t xOff, int16_t yOff,
              bool flipLR, bool flipUD, int16_t rotateDeg)
 {
+    //  This function has been micro optimized by cnlohr on 2022-09-07, using gcc version 8.4.0 (crosstool-NG esp-2021r2-patch3)
+
     if(NULL == wsg->px)
     {
         return;
     }
 
-    // Draw the image's pixels
-    for(int16_t srcY = 0; srcY < wsg->h; srcY++)
+    uint32_t h = disp->h;
+    uint32_t w = disp->w;
+    uint32_t pxmax = w * h;
+    paletteColor_t * px = disp->pxFb;
+    
+    if(rotateDeg)
     {
-        for(int16_t srcX = 0; srcX < wsg->w; srcX++)
+        uint32_t wsgw = wsg->w;
+        uint32_t wsgh = wsg->h;
+        for(int16_t srcY = 0; srcY < wsgh; srcY++)
         {
-            // Draw if not transparent
-            uint16_t srcIdx = (srcY * wsg->w) + srcX;
-            if (cTransparent != wsg->px[srcIdx])
+            int usey = srcY;
+            
+            // Reflect over X axis?
+            if(flipUD)
             {
-                // Transform this pixel's draw location as necessary
-                int16_t dstX = srcX;
-                int16_t dstY = srcY;
-                if(flipLR || flipUD || !rotateDeg)
+                usey = wsg->h - 1 - usey;
+            }
+
+            paletteColor_t * linein = &wsg->px[usey * wsgw];
+            
+            // Reflect over Y axis?
+            int readX = 0;
+            int advanceX = 1;
+            if (flipLR)
+            {
+                readX = wsgw;
+                advanceX = -1;
+            }
+            
+            int16_t localX = 0;
+            for(int16_t srcX = 0; srcX != wsgw; srcX++)
+            {
+                // Draw if not transparent
+                uint8_t color = linein[readX];
+                if (cTransparent != color)
                 {
-                    transformPixel(&dstX, &dstY, xOff, yOff, flipLR, flipUD,
-                                   rotateDeg, wsg->w, wsg->h);
+                    int16_t tx = localX;
+                    int16_t ty = srcY;
+                    rotatePixel(&tx, &ty, rotateDeg, wsgw, wsgh );
```

Comment:
The order used to be rotate -> flipLR -> flipUD, now it seem to be flipUD -> flipLR -> rotate. I don't know if anyone was actually doing multiple operations, but it's worth noting that this code change may also change transformations.